verifier/src/main/java/com/aws/jverify/verifier/Driver.java

```
@@ -287,20 +306,31 @@ private static void parseDafnyJsonOutput(BufferedReader dafnyOutput, Verificatio
                 return;  // nothing to do
             } else if (line.startsWith("{")) {
                 try {
-                    outResults.diagnostics.add(objectMapper.readValue(line, DafnyDiagnostic.class));
+                    DafnyOutput output = objectMapper.readValue(line, DafnyOutput.class);
+                    if (output instanceof DafnyDiagnostic dafnyDiagnostic) {
+                        for (var index = 0; index < dafnyDiagnostic.arguments.length; index++) {
+                            dafnyDiagnostic.arguments[index] = mangler.safeUnmangleName(dafnyDiagnostic.arguments[index]);
+                        }
+                    }
+                    if (output instanceof StatusMessage statusMessage) {
```

Probably good to check that you don't get more than one of these just in case.

---

And then throw an exception? Why would you expect the duplicate message?

---

I don't, but I could imagine a bug in the way we call Dafny leading to duplicates by accident. And I'd rather get an exception rather than just overwriting to only keep the last one.

In fact, you should throw an exception if you get another class other than the two you're handling here too, right?

---

> In fact, you should throw an exception if you get another class other than the two you're handling here too, right?

Good excuse to use Java's sealed and permits feature